src/Model.js

```diff
@@ -318,77 +285,59 @@ export default class Model {
    * too
    */
   @action
-  async destroy ({ optimistic = true }: DestroyOptions = {}): Promise<*> {
-    if (!this.has(this.primaryKey) && this.collection) {
-      this.collection.remove([this.optimisticId])
-      return Promise.resolve()
-    }
+  destroy ({ optimistic = true, ...otherOptions }: DestroyOptions = {}): Request {
+    const collection = this.collection
 
-    const label: Label = 'destroying'
-    const { promise, abort } = apiClient().del(this.url())
-
-    if (optimistic && this.collection) {
-      this.collection.remove([this.id])
+    if (this.isNew && collection) {
+      collection.remove(this)
```

Comment:
ahá @rdiazv :) the reason of all my troubles

Reply:
🤔